### PR TITLE
V2.0.1 mattip

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

- Builds on #76
- Adds me as a maintainer so next time I can push directly to a bot-generated PR
- Removes gpustat restriction since the package is now available on windows
- Removes hard dependency on gym since gym now depends on pygame and ale-py which are not available on conda-forge
- Use matplotlib-base instead of matplotlib in accordance with bot recommendations

### CODE REVIEW

**Comment by mattip** [comment]
@conda-forge-admin, please rerender

**Comment by mattip** [comment]
@conda-forge-admin please rerender

**Comment by mattip** [comment]
Let's see if the macos compilation still suffers from the problems in #2

**Comment by mattip** [comment]
macos is still getting these errors. I thought updating to 10.15 would solve that...:
```
Undefined symbols for architecture x86_64:
  "_objc_loadClassref", referenced from:
      __ARCLite__load() in libarclite_macosx.a(arclite.o)
  "_objc_readClassPair", referenced from:
      __ARCLite__load() in libarclite_macosx.a(arclite.o)
ld: symbol(s) not found for architecture x86_64
```

**Comment by mattip** [comment]
MacOS is failing to compile flatbuffers
```
ERROR: /Users/runner/miniforge3/conda-bld/ray-packages_1667744362102/_build_env/share/bazel/d06f3cbc06c10f16bd4b1ca3b791e8e2/external/com_github_google_flatbuffers/BUILD:74:10: Linking external/com_github_google_flatbuffers/flatc [for host] failed: (Aborted): cc_wrapper.sh failed: error executing command 
  (cd /Users/runner/miniforge3/conda-bld/ray-packages_1667744362102/_build_env/share/bazel/d06f3cbc06c10f16bd4b1ca3b791e8e2/sandbox/darwin-sandbox/32/execroot/com_github_ray_project_ray && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=10.9 \
    PATH=/Users/runner/miniforge3/conda-bld/ray-packages_1667744362102/_build_env/bin:/Users/runner/miniforge3/conda-bld/ray-packages_1667744362102/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin:/Users/runner/miniforge3/condabin:/Users/runner/miniforge3/bin:/Users/runner/miniforge3/condabin:.:/usr/local/bin:/usr/local/sbin:/Users/runner/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/bin:/bin:/usr/sbin:/sbin:/Users/runner/hostedtoolcache/stack/2.9.1/x64 \
    XCODE_VERSION_OVERRIDE=13.2.1.13C100 \
  external/local_config_cc/cc_wrapper.sh @bazel-out/host/bin/external/com_github_google_flatbuffers/flatc-2.params)
Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox
Undefined symbols for architecture x86_64:
  "_objc_loadClassref", referenced from:
      __ARCLite__load() in libarclite_macosx.a(arclite.o)
  "_objc_readClassPair", referenced from:
      __ARCLite__load() in libarclite_macosx.a(arclite.o)
ld: symbol(s) not found for architecture x86_64

```

It seems flatbuffers is pinned to [commit 63d51afd1196336a7d1f56a988091ef05deb1c62](https://github.com/ray-project/ray/blob/b65ec0f85dc060ffeca1a6ae58ddcd26c8f9f066/bazel/ray_deps_setup.bzl#L178) which precedes version 1.11.0 (from April 2019). The current version is [22.10.26](https://github.com/google/flatbuffers/releases/tag/v22.10.26).

**Comment by mattip** [comment]
Linux is passing!

**Comment by mattip** [comment]
On windows, pip could not properly install gym. Rather than fight it I skipped the `rllib` tests. 

<details><summary>pip failure</summary>

```
>python -m pip install gym 
Collecting gym
  Downloading gym-0.26.2.tar.gz (721 kB)
     -------------------------------------- 721.7/721.7 kB 4.6 MB/s eta 0:00:00
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_internal\utils\temp_dir.py", line 36, in global_tempdir_manager
    _tempdir_manager = old_tempdir_manager
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\contextlib.py", line 532, in __exit__
    raise exc_details[1]
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\contextlib.py", line 517, in __exit__
    if cb(*exc_details):
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_internal\utils\temp_dir.py", line 156, in __exit__
    self.cleanup()
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_internal\utils\temp_dir.py", line 173, in cleanup
    rmtree(self._path)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 328, in wrapped_f
    return self(f, *args, **kw)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 408, in __call__
    do = self.iter(retry_state=retry_state)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 364, in iter
    raise retry_exc.reraise()
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 197, in reraise
    raise self.last_attempt.result()
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\concurrent\futures\_base.py", line 439, in result
    return self.__get_result()
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\concurrent\futures\_base.py", line 391, in __get_result
    raise self._exception
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_vendor\tenacity\__init__.py", line 411, in __call__
    result = fn(*args, **kwargs)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\site-packages\pip\_internal\utils\misc.py", line 128, in rmtree
    shutil.rmtree(dir, ignore_errors=ignore_errors, onerror=rmtree_errorhandler)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\shutil.py", line 759, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\shutil.py", line 624, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\shutil.py", line 624, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\shutil.py", line 629, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "D:\bld\ray-packages_1667744425339\_test_env\lib\shutil.py", line 627, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\pip-install-3c5897ge\\gym_c1d1a159b6524853b487a21e7d4be2ce\\gym\\core.py'
```

**Comment by mattip** [comment]
let's try one last time after rerendering with macos 10.11

**Comment by mattip** [comment]
I think this is ready for review.

File: recipe/meta.yaml
Comment (vnlitvinov) [inline recipe/meta.yaml:223]: this basically means `ray-rllib` _package_ in conda-forge is broken... why is using `gym` from conda-forge not enough for it?..

File: recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch
Comment (vnlitvinov) [inline recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch:9]: is there a reason to strip headers?

File: recipe/meta.yaml
Comment (mattip) [reply recipe/meta.yaml:223]: The conda builds of gym are stuck on gym 0.21.0 since versions beyond that depend on pygame and ale-py, which are not yet packaged for conda-forge. We could pin `gym <=21.0`, but then that would preclude #75, since there is no gym for aarch64 and ppc64le. Between `pip install` ing gym and having a aarch64 build, I would think it would be preferable to have an aarch64 build for cloud platforms.

File: recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch
Comment (mattip) [reply recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch:9]: The patch changed significantly, and the information in the header is no longer correct. I wouldn't want the blame for mistakes to be attributed to the original author.

Comment (mattip) [reply recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch]: I restored the removed lines here with a disclaimer that I made subsequent changes.

File: recipe/meta.yaml
Comment (mattip) [reply recipe/meta.yaml]: In rethinking this, it seems better to leave the broken gym as it was for the 2.0.0 release. I think in the long-term the dependency will move to gymnasium, and perhaps it will be easier to get that properly built for conda forge.

I reverted the change from a conda dependency to a pip dependency.

**Review by vnlitvinov — APPROVED** [review]
LGTM assuming tests pass!

Thanks @mattip, would you be also able to try tackle MacOS support? I think requiring 10.15 should be okay compared to alternative (no Ray on MacOS whatsoever) 🙃 , but please open a PR for that and we can discuss with other maintainers.

**Comment by mattip** [comment]
> open a PR for that

OK

**Comment by mattip** [comment]
CI is passing

File: recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch
Comment (h-vetinari) [inline recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch:10]: This line should go into the patch message itself, otherwise it'll be lost upon the next `git format-patch`. You can also just add `Co-Authored-By: ...` - the git history will tell the complete story of how you changed the patch

File: recipe/meta.yaml
Comment (h-vetinari) [inline recipe/meta.yaml:221]: What's the point of changing this away from `imports:`?

File: recipe/meta.yaml
Comment (h-vetinari) [inline recipe/meta.yaml:211]: What do we need pip for?

File: recipe/conda_build_config.yaml
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:19]: Could it be that this hasn't been rerendered yet? The CI files still contain 10.11. Also, we should as a dun-constraint on `__osx` if we raise the minimum target

**Review by h-vetinari — COMMENTED** [review]
Looks good so far, thanks a lot for working on this. A few comments.

File: recipe/conda_build_config.yaml
Comment (mattip) [reply recipe/conda_build_config.yaml:19]: Adding this to the meta.yaml will be part of a follow-on macos-specific PR
```
requirements:
  run:
    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.15") }}  # [osx and x86_64]
```

File: recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch
Comment (mattip) [reply recipe/patches/0003-Remove-all-dependencies-from-setup.py.patch:10]: I admit I do not use `git format-patch` and had originally removed lines 1-11. Another reviewer requested I leave them.

File: recipe/meta.yaml
Comment (mattip) [reply recipe/meta.yaml:221]: It is left over from previous refactorings to make `gym` optional and install it via pip, which is why pip was made part of the runtime dependencies. I will remove pip and revert the testing format.

**Comment by mattip** [comment]
rerendered and cleaned up pip, rllib testing

**Comment by h-vetinari** [comment]
Hey @mattip, sorry for pushing directly into your PR, but I thought this is gonna be quicker than trying to explain myself successfully. The idea is to always be able to apply the patches quickly on a local checkout of the respective source (with the right version), and the patches staying +/- idempotent (most importantly, to persist patch attribution even if you're changing things). I've written down a more [detailed](https://github.com/conda-forge/tensorflow-feedstock/pull/203#pullrequestreview-874021990) mini-guide a while ago if you're interested.

This feedstock has the extra complexity of having a patch-within-a-patch (a hideous thing I'd like to avoid, but alas...), where the same principle would apply, but I didn't actually check out the redis-source now; just made sure that I didn't modify the content of your patch.

**Comment by mattip** [comment]
Thanks. I see there is already a 2.1.0 PR, so it would be nice if we would not fall behind too far.

**Comment by h-vetinari** [comment]
> Thanks. I see there is already a 2.1.0 PR, so it would be nice if we would not fall behind too far.

OK, since @vnlitvinov approved already, I'm giving this an automerge label. :)

**Comment by h-vetinari** [comment]
> Thanks. I see there is already a 2.1.0 PR, so it would be nice if we would not fall behind too far.

Maybe let's get in osx builds before that though? Are you able to take the changes you had already and form them into a PR?

**Comment by mattip** [comment]
Thanks!

**Comment by vnlitvinov** [comment]
> I've written down a more [detailed](https://github.com/conda-forge/tensorflow-feedstock/pull/203#pullrequestreview-874021990) mini-guide a while ago if you're interested.

hey @h-vetinari, I wonder if you could port some of your guide to the document I made here: https://github.com/conda-forge/ray-packages-feedstock/blob/main/ray-patching.md

**Pull request merged**
